src/Native/LibTorchSharp/THSNN.cpp

```diff
@@ -1586,6 +1632,31 @@ Tensor   THSNN_TransformerDecoder_forward(const NNModule module, const Tensor tg
     );
 }
 
+NNModule THSNN_Flatten_ctor(const int64_t start_dim, const int64_t end_dim, NNAnyModule* outAsAnyModule)
+{
+    CATCH_RETURN_NNModule(
+        auto opts = torch::nn::FlattenOptions()
+            .start_dim(start_dim)
+            .end_dim(end_dim);
+
+        auto mod = std::make_shared<torch::nn::FlattenImpl>(opts);
+
+        // Keep a boxed version of the module in case we add it to a Sequential later (the C++ templating means
```

Comment:
If you think it's useful to templatize this repeated code please feel free to do that

Reply:
I'll look into it.